backend-rust/src/graphql_api/transaction_metrics.rs

```diff
@@ -45,61 +45,38 @@ impl QueryTransactionMetrics {
         period: MetricsPeriod,
     ) -> ApiResult<TransactionMetrics> {
         let pool = get_pool(ctx)?;
-
-        let last_cumulative_transaction_count = sqlx::query_scalar!(
-            "SELECT cumulative_num_txs FROM blocks ORDER BY height DESC LIMIT 1"
-        )
-        .fetch_one(pool)
-        .await?;
-
         // The full period interval, e.g. 7 days.
         let period_interval: PgInterval = period
             .as_duration()
             .try_into()
             .map_err(|e| ApiError::DurationOutOfRange(Arc::new(e)))?;
-
-        let cumulative_transaction_count_before_period = sqlx::query_scalar!(
-            "SELECT cumulative_num_txs
-            FROM blocks
-            WHERE slot_time >= (now() - $1::interval)
-            ORDER BY height ASC
-            LIMIT 1",
```

Comment:
This was the query causing issues, since it is `order by height` and filter on `slot_time` postgres could not optimize this properly.
Instead it should have been `order by slot_time`, but since the information is part of the rows below it can be skipped entirely.